Non-blocking functions

adnan-ashraf opened the pull request:
Provided functions enabling non-blocking operation (closes #49)

adnan-ashraf commented:
Can someone please review this pull request? @Letme

Letme commented on src/mlx90632.c:
```diff
@@ -44,18 +44,25 @@ static const char mlx90632version[] __attribute__((used)) = { VERSION };
+
+int32_t mlx90632_wait_for_measurement(void)
+{
+    int32_t ret;
+    uint16_t reg_status;
+    int tries = MLX90632_MAX_NUMBER_MESUREMENT_READ_TRIES;
```
lets arrange variables in inverted xmass tree:

```suggestion
    int tries = MLX90632_MAX_NUMBER_MESUREMENT_READ_TRIES;
    uint16_t reg_status;
    int32_t ret;
```

Letme commented on src/mlx90632.c:
```diff
@@ -577,20 +601,22 @@ int32_t mlx90632_start_measurement_burst(void)
+
+int32_t mlx90632_wait_for_measurement_burst(void)
+{
+    int32_t ret;
+    uint16_t reg_status;
+    int tries = MLX90632_MAX_NUMBER_MESUREMENT_READ_TRIES;
```
```suggestion
    int tries = MLX90632_MAX_NUMBER_MESUREMENT_READ_TRIES;
    uint16_t reg_status;
    int32_t ret;
```

Letme commented on src/mlx90632.c:
```diff
@@ -608,6 +634,42 @@ int32_t mlx90632_start_measurement_burst(void)
+
+int32_t mlx90632_trigger_measurement_single(void)
+{
+    int32_t ret;
+    uint16_t reg;
```
```suggestion
    uint16_t reg;
    int32_t ret;
```

Letme commented on src/mlx90632.c:
```diff
@@ -704,4 +766,16 @@ mlx90632_meas_t mlx90632_get_refresh_rate(void)
 
+int32_t mlx90632_get_channel_position(void)
+{
+    int32_t ret;
+    uint16_t reg_status;
```
```suggestion
    uint16_t reg_status;
    int32_t ret;
```

Letme reviewed (changes requested):
Nice contribution. Thanks for the effort. I just have few stylistic requests.

Sorry it took so long, I must have missed this somehow.

Letme commented:
Ok, now CI is also running. Can I ask you to also add some tests for your changes, to make sure we do not break anything (and that we did not break anything now)?

Letme commented:
Regarding CI: I need to fix master, but I am waiting for Ceedling 1.0.0 release (this current version in master is fairly close to the 1.0.0, so it should be easy). That means that currently unit tests are not running as we are referencing non-existing pre-release https://github.com/ThrowTheSwitch/Ceedling/releases

adnan-ashraf commented:
Thanks for taking the time to review!

Appreciate the feedback! Your suggestions have been added.

Sure, I'll write the tests.

yogeshiggalore commented:
when this will get merged..?

Letme commented:
As soon as some tests are added to maintain functionality. I will do a local run with a latest pre-release to make sure that after fix of CI we have all green checks.

Before for some reason CI was not run on external pull requests, but now we fixed this. The only problem is that Ceedling did not release 1.0.0 version yet.

adnan-ashraf commented:
I've added the unit tests, please review the changes.

I was able to run the tests successfully only when I changed `0UL` to `0ULL` in `GENMASK` definition, or by changing `BITS_PER_LONG` to `32` in project.yml file. Should we create another issue and pull request to fix `GENMASK` definition, or should I go ahead with the fix in this branch? Please suggest!

adnan-ashraf commented:
I've fixed the `GENMASK` definition making it independent of `BITS_PER_LONG`, and also fixed some other issues. In addition, I've also improved the tests code. Please review the changes and provide any feedback, thanks!

Letme commented:
Can you also run stylechecker, as it seems uncrusitfy complains about something?

Letme reviewed (approved):
Thanks for the complete contribution. I do not have the environment ready to do a complete temperature range check, I will rely on your and my manual tests. Once I fix ceedling with released gem also the CI will maintain the quality.
I think stylechecker complains about a line length, but that should be a quick run fix and then we merge.

adnan-ashraf commented on inc/mlx90632.h:
```diff
@@ -61,15 +61,15 @@
 #endif
 #define GENMASK(h, l) \
-    (((~0UL) << (l)) & (~0UL >> (BITS_PER_LONG - 1 - (h))))
+    ((((1U << h) - 1) | (1U << h)) & ~((1U << l) - 1))
```
Tested and worked on both 32-bits and 64-bits based systems

adnan-ashraf commented:
I've run uncrustify, so the stylechecker is working successfully now!

adnan-ashraf commented:
I'm sorry that stylechecker is complaining again. I used uncrustify latest version 0.79.0 and it locally ran successfully. The stylechecker workflow is using uncrustify version 0.75.1. This could be the reason of failure.

adnan-ashraf commented:
I've updated the stylechecker to use the latest version of uncrustify. I think, it shouldn't complain now. Thanks!

Letme commented:
Great. Newer uncrustify is also one less worry for me to update. Thanks for the initiative.

adnan-ashraf commented:
Glad to hear, happy to help!
